rotkehlchen/premium/premium.py

```diff
@@ -514,10 +566,39 @@ def premium_create_and_verify(credentials: PremiumCredentials, username: str) ->
     - RemoteError if there are problems reaching the server
     """
     premium = Premium(credentials=credentials, username=username)
-    premium.query_last_data_metadata()
+    premium.authenticate_device()
     return premium
 
 
 def has_premium_check(premium: Premium | None) -> bool:
     """Helper function to check if we have premium"""
     return premium is not None and premium.is_active()
+
+
+def get_user_limit(premium: Premium | None, limit_type: USER_LIMIT_TYPE) -> tuple[int, bool]:
+    """Helper function to get a specific user limit and premium status
+
+    Returns:
+        tuple[int, bool]: (limit_value, has_premium)
+    """
+    if premium is None or premium.is_active() is False:
+        log.debug(f'No premium subscription or inactive, returning free limit for {limit_type}')
+        return _get_free_limit(limit_type), False
+
+    try:
+        limits = premium.fetch_limits()
+        return limits[limit_type], True
+    except (RemoteError, PremiumAuthenticationError) as e:
+        log.error(f'Failed to fetch limits from server: {e}. Falling back to free limits')
+        return _get_free_limit(limit_type), False
+
+
+def _get_free_limit(limit_type: USER_LIMIT_TYPE) -> int:
+    """Get the free limit for a specific limit type"""
+    if limit_type == 'history_events_limit':
+        return FREE_HISTORY_EVENTS_LIMIT
+    if limit_type == 'pnl_events_limit':
+        return FREE_PNL_EVENTS_LIMIT
+
+    # can only be 'reports_lookup_limit'
+    return FREE_REPORTS_LOOKUP_LIMIT
```

Comment:
I believe it would be better to have some exhaustive check here because I see us adding a new limit and forgetting to update it here